quiz/models.py

```diff
@@ -26,13 +28,29 @@ class Room(models.Model):
     # Room name, as supplied by user
     name = models.CharField(max_length=200)
     # Unique room code
-    code = models.SlugField(unique=True)
+    code = models.SlugField(unique=True, max_length=5)
     # Variable number of questions in a room
     questions = models.ManyToManyField(Question)
 
     def __str__(self):
         return self.name
 
+    def code_generator(self, size=5):
+        chars = string.ascii_lowercase + string.digits
+        return ''.join(random.choice(chars) for _ in range(size))
+
+    def save(self, *args, **kwargs):
+        # Generate a unique code for the room
+        if not self.code:
+            self.code = self.code_generator()
+        # Save the room. If the code clashes generate a new one and try again.
+        while True:
+            try:
+                with transaction.atomic():
+                    return super(Room, self).save(*args, **kwargs)
+            except IntegrityError:
+                self.code = self.code_generator()
```

Comment:
If we have ran out of rooms, this will run indefinitely, correct? Would that load slow down the PostgreSQL at all?

Reply:
This will run infinitely as you said. The chances of collision are minute, never mind running our altogether.
There are 36^5 possible rooms
What do you think we should do?

Reply:
I agree that this is so unlikely to happen that we can ignore it for now (although maybe a future issue should be opened to mention it). It'll be fine until Qwiz has millions of rooms!